packages/@tailwindcss-upgrade/src/codemods/migrate-border-compatibility.test.ts

```diff
@@ -0,0 +1,235 @@
+import { __unstable__loadDesignSystem } from '@tailwindcss/node'
+import dedent from 'dedent'
+import postcss from 'postcss'
+import { expect, it } from 'vitest'
+import { formatNodes } from './format-nodes'
+import { migrateBorderCompatibility } from './migrate-border-compatibility'
+
+const css = dedent
+
+async function migrate(input: string) {
```

Comment:
Let's add a test here as well with a `Config` that does define `borderColor.DEFAULT` and one where said value uses a `theme()` function. I saw code related to it in your implementation 🙂

Reply:
In this file it's a bit tricky since we resolve user configs, so I added dedicated tests for all of these in the integration tests:
https://github.com/tailwindlabs/tailwindcss/blob/e6c13b968986f0469fe7d67aa78c4d2c213978be/integrations/upgrade/js-config.test.ts#L646

The backwards compatibility CSS uses the `theme(…)` function, which gets replaced which you can also see in that file above.

I can of course add some tests to this file as well where I mock the user config. But it was easier to write integration tests for.

Reply:
oh I skipped over this file since there were so many diffs because of the new comments 🙈 Thanks!

Reply:
Thank you GitHub for hiding the diff 😂